kosat-core/src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -332,10 +329,7 @@ class CDCL {
                 lemma.lbd = lemma.distinctBy { vars[it.variable].level }.size
 
                 // return to decision level where lemma would be propagated
-                backjump(lemma)
-
-                // after backjump there is only one clause to propagate
-                qhead = trail.size
+                cancelUntil(if (lemma.size > 1) vars[lemma[1].variable].level else 0)
```

Comment:
Consider `val level = ...; cancelUntil(level)`: extracting a variable is more clear.